### PR TITLE
list `numPUs` as an exeption to capitalization rule

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
In this poll: https://github.com/chapel-lang/chapel/issues/20445#issuecomment-1238189331, it was decided to list `locale.numPUs` as an exception to the acronym capitalization convention discussed here: https://github.com/chapel-lang/chapel/issues/20240.

A new list of exceptions was started on the style-guide documentation page. `numPUs` is currently the only entry. 

I used the convention description from this PR: https://github.com/chapel-lang/chapel/pull/20392, which has not been merged yet.

### CODE REVIEW

**Review by riftEmber — APPROVED** [review]
lgtm (but should be merged after #20392)

**Comment by mppf** [comment]
@jeremiah-corrado - I've just merged the other PR, FWIW.

File: doc/rst/developer/bestPractices/StandardModuleStyle.rst
Comment (bradcray) [inline doc/rst/developer/bestPractices/StandardModuleStyle.rst:38]: I didn't see this PR before it got merged, but I might be inclined to put a statement here along the lines of.  "Our hope and intention is to update our rules over time so that they cover cases like this, obviating the need to call it out as an exception."  (i.e., indicate that we're not content to have exceptions).  If we thought we needed it we could say "At this time, we don't yet know which of several candidate rules might be used to explain the exception." — but maybe that's expressing a bit more doubt than we're comfortable with?).

Comment (jeremiah-corrado) [reply doc/rst/developer/bestPractices/StandardModuleStyle.rst:38]: I did not think of that, but I like the idea.

What about: "Our hope and intention is to update our rules over time so that they cover cases like this, obviating the need to call it out as an exception. As more exceptions accumulate, rules may be amended to cover multiple exceptions at once" ?

Comment (bradcray) [reply doc/rst/developer/bestPractices/StandardModuleStyle.rst:38]: That sounds fine to me.  We could also have @riftEmber give it a look as a third pair of eyes and your original reviewer.

Comment (riftEmber) [reply doc/rst/developer/bestPractices/StandardModuleStyle.rst:38]: Yes, I would be in favor of that wording.

Comment (jeremiah-corrado) [reply doc/rst/developer/bestPractices/StandardModuleStyle.rst:38]: Great, I'll create an update for this PR with that change.

**Pull request merged**
